### PR TITLE
NodeJS-compatible modules

### PR DESCRIPTION
**Pull request by bhollis**
Had to read through https://nodejs.org/api/esm.html carefully to figure this out, but this should let us use bungie-api-ts from newer NodeJS versions.

Major changes:

1. Mark the package as `type: module`.
2. Switch to including `.js` at the end of imports because that's how Node wants it.
3. Generate `package.json` as part of the generator script. As part of this, I'm generating the new `exports` key. See https://nodejs.org/api/esm.html#esm_subpath_exports, but this is apparently the only way to let NodeJS import sub-paths.
1. No more typedoc. Nobody cared about it.
2. Included the new manifest helpers in the `destiny2` export.

I haven't committed the generated code yet, in order to make this easier to review.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,33 +1,30 @@
...
 cp generator/http.ts generated-src
 
 # Compile and run the generator
 tsc -p tsconfig.json
-node ./build/generate.js
-
-rm -rf ./docs
-typedoc --target ES5 --out ./docs/ --readme README.md generated-src
-touch ./docs/.nojekyll
+node --experimental-json-modules ./build/generate.js
```
Comment (sundevour) [inline build.sh:13]:
```
node.exe: --experimental-json-modules requires --experimental-modules be enabled
```
maybe due to my version?

File: generator/generate-package-json.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import packageJson from './package.json';
```
Comment (sundevour) [inline generator/generate-package-json.ts:1]: could you just make this a `..` and save yourself a `cp` in `build.sh` line 4?
or am i missing a pitfall here

File: README.md [hunk 3]
```diff
@@ -57,6 +57,23 @@ import { Destiny2 } from 'bungie-api-ts';
...
+
+const destinyManifest = await getDestinyManifest($http);
+const manifestTables = getDestinyManifestSlice($http, {
+  destinyManifest,
+  tableNames: ['DestinyInventoryItemDefinition', 'DestinySocketDefinition'];
+  language: 'en';
+});
+
+// manifestTables is an object with properties DestinyInventoryItemDefinition and DestinySocketDefinition
+```
```
Comment (sundevour) [inline README.md:75]: this section looks good, thanks

File: generator/generate-index.ts [hunk 4]
```diff
@@ -11,14 +11,19 @@ export function generateIndex(
...
+export * from './api.js';`;
 
   if (componentsByFile[`${tag.toLowerCase()}/interfaces.ts`]) {
-    imports = `${imports}\nexport * from './interfaces';`;
+    imports = `${imports}\nexport * from './interfaces.js';`;
+  }
+
+  // Destiny2 service has special manifest helpers
+  if (tag === 'destiny2') {
+    imports = `${imports}\nexport * from './manifest.js';`;
```
Comment (sundevour) [inline generator/generate-index.ts:26]: this interrupt doesn't seem to take effect

File: build.sh (hunk 1 above)
Comment (bhollis) [reply build.sh:13]: The `type: module` thing in package.json automatically implies `--experimental-modules`

File: generator/generate-package-json.ts (hunk 2 above)
Comment (bhollis) [reply generator/generate-package-json.ts:1]: Typescript is being picky and won't let you reference something out of `rootDir`

**Pull request merged**